### PR TITLE
Griffin ABTR XS generation and steady-state analysis workflow

### PR DESCRIPTION
**Pull request by shikhar413**
refs #386. Still needs testing to be added

This model will be the starting point for an upcoming hands-on SFR training led by the Griffin team

### CODE REVIEW

**Comment by shikhar413** [comment]
@GiudGiud hoping to contribute this model to the VTB. However, this requires the griffin_data submodule within the Griffin repository to be available (https://github.inl.gov/ncrc/griffin_data). Is there a way to access this repository when running CIVET testing?

**Comment by GiudGiud** [comment]
We'll need to modify the civet recipes to pull it on testing.

**Comment by shikhar413** [comment]
@GiudGiud this PR should be ready for review now. I've added a tests file but I think the tests will fail until the path to griffin_data becomes available on the CIVET machines. Also, what are the steps I can take to run civet tests for virtual_test_bed repo locally?

Note that `MCC3CrossSection/griffin_data` in `xs_generation.i` takes in the path to the griffin_data submodule, so once this is available I can update the tests for use in CIVET.

For HPC runtime, 2 nodes on Sawtooth should be sufficient, and it takes ~15 minutes for the cross section generation and ~40minutes for the full core solve.

**Comment by shikhar413** [comment]
> Also, what are the steps I can take to run civet tests for virtual_test_bed repo locally?

Never mind, figured it out

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (GiudGiud) [inline doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:47]:
```suggestion
This section focuses on the input files needed to run the cross section generation worfklow. Two input files are called for this section in order to split mesh generation from the actual execution of the cross section generation workflow calculations. `xs_generation_mesh.i` defines the input heterogeneous mesh through [!ac](RGMB) mesh generators and generates the equivalent R-Z mesh that will be used as part of the Griffin R-Z spectrum calculation, while `xs_generation.i` imports this mesh through [FileMeshGenerator](https://mooseframework.inl.gov/source/meshgenerators/FileMeshGenerator.html) and defines all control parameters related to the cross section generation workflow. The commands to run the cross section generation workflow from start to finish are shown below:
```

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (GiudGiud) [inline doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:58]:
```suggestion
The [ReactorMeshParams](https://mooseframework.inl.gov/source/meshgenerators/ReactorMeshParams.html), [PinMeshGenerator](https://mooseframework.inl.gov/source/meshgenerators/ReactorMeshParams.html), [AssemblyMeshGenerator](https://mooseframework.inl.gov/source/meshgenerators/AssemblyMeshGenerator.html), and [CoreMeshGenerator](https://mooseframework.inl.gov/source/meshgenerators/CoreMeshGenerator.html) in `xs_generation_mesh.i` make use of the [!ac](RGMB) mesh generation system within the Reactor module to streamline the definition of latticed core geometries and the reporting IDs associated with such structures [!citep](shemon2023reactor). Mesh generation through [!ac](RGMB) is a requirement for taking advantage of Griffin's [!ac](SFR) cross section generation workflow, as it ensures that all reporting IDs related to region ID mappings as well as IDs corresponding to groupings of pin structures, assembly structures, and axial layers are defined in a consistent and predictable manner.
```

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (GiudGiud) [inline doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:190]:
```suggestion
`MCC3CrossSection/map_het_grid_name`, `MCC3CrossSection/map_het_grid_ref_value`, and `MCC3CrossSection/map_het_grid_values` control how cross sections are tabulated as a function of tabulation variables such as temperature. `MCC3CrossSection/map_het_grid_name` maps each composition defined in the `[Compositions]` block that has an associated region ID to an associated grid variable. For the purposes of this tutorial, each heterogeneous region has a single temperature variable it is tabulated against, where `Tfuel` represents the fuel temperature, and `Tcool` represents the coolant temperature. `MCC3CrossSection/map_het_grid_ref_value` sets the reference value for each grid variable of interest, while `MCC3CrossSection/map_het_grid_values` defines all tabulated values that are assocaited with each grid variable.
```

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (GiudGiud) [inline doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:214]: use a listing this wont age well

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (GiudGiud) [inline doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:309]: same

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (GiudGiud) [inline doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:297]: I wonder if you could
!include that input file instead, with the right 'active/inactive' parameter in [Mesh]
or maybe with a final_generator_name

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Comment by shikhar413** [comment]
> Griffin testing should run now. let s see

Thanks for the additions!

File: doc/config.yml
Comment (shikhar413) [inline doc/config.yml:34]: @GiudGiud looks like this is causing the documentation test to fail

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (shikhar413) [reply doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:305]: @GiudGiud in case it wasn't clear through the documentation, Griffin will automatically generate a `core_hom_macro.i` file that will be the starting point for running the full core solve step. The user will have to modify this file slightly to get it to actually run for this problem, so I'm wondering if it makes sense to just commit a file `core_hom_macro_updated.i` that has all the additions needs for the full core solve step to properly run. Previously I was trying to do this all through cli args in order to make sure that the process of file generation and file editing could be tested properly, but the downside of this is that the documentation page does not have a file `core_hom_macro.i` to reference. What do you suggest?

File: doc/config.yml
Comment (GiudGiud) [reply doc/config.yml:34]: ah too early then. Let s remove it

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (shikhar413) [reply doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:305]: The other way this could potentially be simplified is to define another file `abtr_het_mesh.i` which contains the shared mesh blocks between `xs_generation_mesh.i` and `core_hom_macro.i`, since the het mesh generation step is exactly the same in both cases.

File: doc/config.yml
Comment (shikhar413) [reply doc/config.yml]: Removed

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (shikhar413) [reply doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:305]: Basically, the slightly confusing thing here is explaining what changes are needed to `core_hom_macro.i` to define the Mesh block. Mesh blocks needed to be added to both the beginning (het mesh definition) and the end of the mesh block (coarse mesh for equivalent core) in `core_hom_macro.i`

Comment (shikhar413) [reply doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md]: @GiudGiud I've gone ahead and added the file `abtr_het_mesh.i`, which has all the meshing routines that are common to both xs generation and full core solve steps. The only block in the documentation that is manually defined is the `[coarse_mesh]` block, since this is the only block that is specific to the full core solve step. Let me know if this is sufficient

File: sfr/abtr_xsgen_workflow/abtr_het_mesh.i
Comment (GiudGiud) [inline sfr/abtr_xsgen_workflow/abtr_het_mesh.i:2]:
```suggestion
# ABTR Equivalent-full core heterogeneous mesh input
```

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (GiudGiud) [inline doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:261]: a list of the changes would be more readable?
```suggestion
The command line options specify that:

 - mesh refinement is applied
 - `EquivalentCoreMeshGenerator` settings are updated through the `[Mesh]` block
 - the boundary conditions sidesets are set
 - DFEM-SN angular discretizations are updated to increase solution accuracy through the `[TransportSystems]` block
 - CMFD settings are updated through the `[Executioner]` block
 - the reactor power is set through the `[PowerDensity]` block.
 
These parameter settings ensure that calculated eigenvalues are predicted to within 100pcm of previous modeling work for the [!ac](ABTR) reactor core [!citep](pyarc_report).
```

File: sfr/abtr_xsgen_workflow/core_hom_macro.i
Comment (GiudGiud) [inline sfr/abtr_xsgen_workflow/core_hom_macro.i:9]:
```suggestion
# This input file is automatically generated by the Griffin EquivalentCoreGenerator
# Input files modifications that MUST be done by the user:
```

File: sfr/abtr_xsgen_workflow/core_hom_macro.i
Comment (GiudGiud) [inline sfr/abtr_xsgen_workflow/core_hom_macro.i:16]:
```suggestion
# See online documentation or 'tests' file for the command line arguments to pass to do this
```

File: sfr/abtr_xsgen_workflow/xs_generation.i
Comment (GiudGiud) [inline sfr/abtr_xsgen_workflow/xs_generation.i:9]: could you do this
```suggestion
!include xs_generation_mesh.i

[MCC3CrossSection]
```

File: sfr/abtr_xsgen_workflow/xs_generation_mesh.i
Comment (GiudGiud) [inline sfr/abtr_xsgen_workflow/xs_generation_mesh.i:8]: and this?
```suggestion
!include !abtr_het_mesh.i
```
or not?

seems easier than the multiple input files

**Review by GiudGiud — CHANGES_REQUESTED** [review]
good to go after this

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (shikhar413) [reply doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:176]: By original heterogeneous RZ mesh, do you mean an RZ mesh that represents pin-level heterogeneity? If so, then no. The RZ mesh is based on an assembly homogenized lattice configuration

Comment (GiudGiud) [reply doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:176]: ok then I like the amended better. If you want more detail that works too

File: sfr/abtr_xsgen_workflow/xs_generation_mesh.i
Comment (shikhar413) [reply sfr/abtr_xsgen_workflow/xs_generation_mesh.i:8]: Cool, this works. Let me update to this

Comment (shikhar413) [reply sfr/abtr_xsgen_workflow/xs_generation_mesh.i:8]: I'd also need to update the documentation to describe the updated run command and that this command just imports the referenced .i file right?

File: sfr/abtr_xsgen_workflow/core_hom_macro.i
Comment (shikhar413) [reply sfr/abtr_xsgen_workflow/core_hom_macro.i]: Updated to:

# This input file is automatically generated by the Griffin [MCC3CrossSection] block when generating the 33 group SFR cross section library

File: doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md
Comment (shikhar413) [reply doc/content/sfr/abtr_xsgen_workflow/abtr_xsgen_workflow.md:239]: @GiudGiud this is not part of any input file listing right now, since core_hom_macro.i does not define a coarse mesh. Should I add these parameters in the CLI section where I discuss how to run the problem?

File: sfr/abtr_xsgen_workflow/xs_generation_mesh.i
Comment (shikhar413) [reply sfr/abtr_xsgen_workflow/xs_generation_mesh.i:8]: Documentation is updated to describe this process

**Comment by shikhar413** [comment]
@GiudGiud all comments should be addressed now, except for the one about using a listing for the full core solve coarse mesh, since this mesh is not automatically generated by `core_hom_macro.i`. Let me know if you'd prefer to define that mesh block in the CLI portion of the documentation instead.

**Comment by GiudGiud** [comment]
I ll take a look this could be fixable

**Comment by GiudGiud** [comment]
ok fixable. I m pushing a patch to MOOSE. I ll merge this once the patch is in

**Comment by shikhar413** [comment]
Sounds good, thanks for taking care of this

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
@milljm is helping us update pyhit with my change. Once it is in, an invalidate and it's good to go. i can click that from Japan or @mtnwalker, @kyleeswanson and @nathangrund have merge rights

**Comment by shikhar413** [comment]
@milljm any update on this? I'm hoping to get a stable link to submit for EC this week for an upcoming workshop that the Griffin team is organizing that would be referencing the documentation to this tutorial, thanks!

**Pull request merged**
